### PR TITLE
Add CreateRequest Activity and connect the whole flow

### PR DESCRIPTION
**Pull request by assafgelber**

### CODE REVIEW

File: app/src/main/java/com/iic/shopingo/ui/request_flow/Activities/CreateRequestActivity.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import butterknife.OnClick;
+import com.iic.shopingo.R;
+import com.iic.shopingo.ui.request_flow.views.CreateRequestItemListView;
+import com.iic.shopingo.ui.request_flow.views.CreateRequestListItemView;
+import java.util.Currency;
+import java.util.Locale;
+
+public class CreateRequestActivity extends ActionBarActivity implements
+    CreateRequestListItemView.OnListViewChanged {
+  public static final String REQUEST_EXTRA_KEY = "request";
```
Comment (ifeins) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/Activities/CreateRequestActivity.java]: let's prefix those constants with EXTRA, EXTRA_REQUEST_KEY.
This way it's easy to find all of them in auto complete.

File: app/src/main/java/com/iic/shopingo/ui/request_flow/Views/CreateRequestListItemView.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  public static CreateRequestListItemView inflate(ViewGroup parent) {
+    CreateRequestListItemView itemView = (CreateRequestListItemView) LayoutInflater.from(parent.getContext())
+        .inflate(R.layout.create_request_list_item, parent, false);
+    return itemView;
+  }
+
+  private void init() {
+    LayoutInflater.from(getContext()).inflate(R.layout.create_request_list_item_children, this, true);
```
Comment (ifeins) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/Views/CreateRequestListItemView.java]: any reason why it's not `create_request_list_item`, why the `children` suffix?

File: app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java [hunk 3]
```diff
@@ -20,27 +23,36 @@
...
+    startActivity(intent);
   }
 
   @Optional
   @OnClick(R.id.request_state_settle_button)
   public void onSettleRequest(View view) {
-    // TODO: Settle request in server and go home
+    // TODO: Settle request in server
+    Intent intent = new Intent(this, HomeActivity.class);
+    startActivity(intent);
```
Comment (ifeins) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java:47]: after `startActivity` you need to call `finish`, as you don't want to allow the user back to return him to this activity.

File: app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java [hunk 4]
```diff
@@ -20,27 +23,36 @@
...
   }
 
   @Optional
   @OnClick(R.id.request_state_try_again_button)
   public void onTryAgain(View view) {
-    // TODO: Go to trippers list
+    request.shopper = null;
+    Intent intent = new Intent(this, SelectShopperActivity.class);
+    intent.putExtra(SelectShopperActivity.REQUEST_EXTRA_KEY, request);
+    startActivity(intent);
```
Comment (ifeins) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java:57]: finish()

File: app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java (hunk 3 above)
Comment (agelber) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java]: Why not?

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -12,11 +12,19 @@
...
   <string name="request_state_try_again_button">Try someone else</string>
   <string name="request_state_go_yourself_button">Go yourself</string>
-  
+
+  <!-- Create Request -->
+  <string name="title_activity_create_request">CreateRequestActivity</string>
+  <string name="create_request_new_item_hint">Add another item</string>
+  <string name="create_request_remove_item_button">X</string>
+  <string name="create_request_price_hint">5.00</string>
+  <string name="create_request_create_button">Create Request</string>
```
Comment (ifeins) [inline app/src/main/res/values/strings.xml]: extra to its own file `strings_request_flow`

**Comment by ifeins** [comment]
:cool:

File: app/src/main/java/com/iic/shopingo/ui/request_flow/Views/CreateRequestListItemView.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+  @OnClick(R.id.create_request_remove_button)
+  public void onRemoveClick(View view) {
+    ((OnListViewChanged) getContext()).onRemoveButtonClicked(this);
+  }
+
+  @OnFocusChange(R.id.create_request_item_title)
+  public void onTitleFocusChanged(EditText view, boolean hasFocus) {
+    if (!hasFocus) {
+      ((OnListViewChanged) getContext()).onItemEdited(this, view.getText().toString());
```
Comment (AsafGartner) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/Views/CreateRequestListItemView.java]: This is not a good idea.
You're assuming that the context that you get is the parent activity (which doesn't have to be the case), and then you assume that it implements `OnListViewChanged`.
It's better to have `setOnListViewChangedListener` and the activity should call it.

File: app/src/main/java/com/iic/shopingo/ui/request_flow/Views/CreateRequestListItemView.java (hunk 2 above)
Comment (ifeins) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/Views/CreateRequestListItemView.java:56]: I just went over the blog post you sent me and I see no reason for having two layouts, you can nest the EditText and Button in the `R.layout.create_request_list_item` layout. Someone even ask him that in the blog post's comments.

Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/Views/CreateRequestListItemView.java:56]: It would just mean that it wouldn't be reusable, no?

**Pull request merged**